src/rdb.c

```diff
@@ -303,18 +302,18 @@ void *rdbLoadIntegerObject(rio *rdb, int enctype, int flags, size_t *lenptr, siz
         rdbReportCorruptRDB("Unknown RDB integer encoding type %d",enctype);
         return NULL; /* Never reached. */
     }
-    if (plainFlag || sdsFlag || hfldFlag) {
+    if (plainFlag || sdsFlag) {
         char buf[LONG_STR_SIZE], *p;
         int len = ll2string(buf,sizeof(buf),val);
         if (lenptr) *lenptr = len;
         if (plainFlag) {
             p = zmalloc_usable(len, usable);
-        } else if (sdsFlag) {
+        } else {
+            serverAssert(sdsFlag);
```

Comment:
nit: Switch `debugServerAssert` for consistency with the similar `debugServerAssert`s below?